core/src/helpers.rs

```diff
@@ -169,16 +172,281 @@ pub fn starknet_block_to_eth_block(block: MaybePendingStarknetBlock) -> RichBloc
+            ether_tx.from = starknet_address_to_ethereum_address(declare_tx.sender_address);
+            // Define gas_price data
+            ether_tx.gas_price = None;
+            // Extracting the signature
+            ether_tx.r = field_element_to_u256(declare_tx.signature[0]);
+            ether_tx.s = field_element_to_u256(declare_tx.signature[1]);
+            ether_tx.v = field_element_to_u256(declare_tx.signature[2]);
+            // Extracting the to address
+            ether_tx.to = None;
+            // Extracting the value
+            ether_tx.value = U256::from(0);
+            // Extracting the gas
+            ether_tx.gas = U256::from(0);
+            // Extracting the chain_id
+            ether_tx.chain_id = None;
+            // Extracting the standard_v
+            ether_tx.standard_v = U256::from(0);
+            // Extracting the public_key
+            ether_tx.public_key = None;
+        }
+        StarknetTransaction::Deploy(deploy_tx) => {
+            // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+            ether_tx.hash = H256::from_slice(&deploy_tx.transaction_hash.to_bytes_be());
+            // Define gas_price data
+            ether_tx.gas_price = None;
+
+            ether_tx.creates = None;
+            // Extracting the public_key
+            ether_tx.public_key = None;
+        }
+        StarknetTransaction::DeployAccount(deploy_account_tx) => {
+            ether_tx.hash = H256::from_slice(&deploy_account_tx.transaction_hash.to_bytes_be());
+            ether_tx.nonce = field_element_to_u256(deploy_account_tx.nonce);
+            // TODO: Get from estimate gas
+            ether_tx.gas_price = None;
+            // Extracting the signature
+            ether_tx.r = field_element_to_u256(deploy_account_tx.signature[0]);
+            ether_tx.s = field_element_to_u256(deploy_account_tx.signature[1]);
+            ether_tx.v = field_element_to_u256(deploy_account_tx.signature[2]);
+            // Extracting the to address
+            ether_tx.to = None;
+            // Extracting the gas
+            ether_tx.gas = U256::from(0);
+            // Extracting the chain_id
+            ether_tx.chain_id = None;
+            // Extracting the standard_v
+            ether_tx.standard_v = U256::from(0);
+            // Extracting the public_key
+            ether_tx.public_key = None;
+        }
+    }
+
+    ether_tx
+}
+
+fn field_element_to_u256(x: FieldElement) -> U256 {
+    let inner: u64 = x.to_string().parse().unwrap();
+    U256::from(inner)
+}
+
+fn vec_felt_to_bytes(contract_bytecode: Vec<FieldElement>) -> Bytes {
+    let contract_bytecode_in_u8: Vec<u8> = contract_bytecode
+        .into_iter()
+        .flat_map(|x| x.to_bytes_be())
+        .collect();
+    Bytes::from(contract_bytecode_in_u8)
+}
+
+fn starknet_address_to_ethereum_address(x: FieldElement) -> Address {
+    let inner: u64 = x.to_string().parse().unwrap();
```

Comment:
let's not unwrap:)!

also, it's probably safer to go from felt to bytes then Address from Bytes:)!

Reply:
Didnt get the second comment, this function is to get from Felt to Address, so got a bit confused

Reply:
Something like `let inner: H160 = H160::from_slice(x.to_bytes_be());` has lesser chances to panic, as u64 is 64 bits integer, if x > u64::MAX(), code will panic